lib/ansible/modules/cloud/amazon/s3.py

```diff
@@ -713,6 +713,13 @@ def main():
 def get_s3_connection(aws_connect_kwargs, location, rgw, s3_url):
     if s3_url and rgw:
         rgw = urlparse(s3_url)
+        # ensure none of the named arguments we will pass to boto.connect_s3 
+        # are already present in aws_connect_kwargs
+        for kw in ['is_secure', 'host', 'port', 'calling_format']:
```

Comment:
I don't really like manually setting aws_connect_kwargs in this function. And none of these except calling_format are ever set in main(). I don't understand why it's necessary to manually delete these.